ground/src/main/java/com/google/android/ground/ui/datacollection/components/LoiNameDialog.kt

```diff
@@ -0,0 +1,129 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.ButtonColors
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldColors
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.em
+import androidx.core.content.ContextCompat
+import com.google.android.ground.R
+import com.google.android.ground.ui.common.AbstractFragment
+import com.google.android.material.color.MaterialColors
+
+private fun AbstractFragment.getColor(id: Int): Color =
+  Color(ContextCompat.getColor(requireContext(), id))
+
+private fun AbstractFragment.getMaterialColor(id: Int): Color =
+  Color(MaterialColors.getColor(requireContext(), id, ""))
+
+@Composable
+private fun AbstractFragment.getElementColors(): Triple<ButtonColors, Color, TextFieldColors> {
+  val primaryColor = getMaterialColor(R.attr.colorPrimary)
+  val onPrimaryColor = getColor(R.color.md_theme_onPrimary)
+  val onSurfaceDisabledColor = getColor(R.color.md_theme_on_surface_disabled)
+  val textFieldColor = getColor(R.color.md_theme_textFieldContainers)
+  val saveButtonColors =
+    ButtonColors(
+      containerColor = primaryColor,
+      contentColor = onPrimaryColor,
+      disabledContainerColor = onSurfaceDisabledColor,
+      disabledContentColor = onPrimaryColor,
+    )
+  val cancelButtonColor = getMaterialColor(R.attr.colorPrimary)
+  val textFieldColors =
+    TextFieldDefaults.colors(
+      focusedIndicatorColor = primaryColor,
+      unfocusedIndicatorColor = primaryColor,
+      focusedContainerColor = textFieldColor,
+      unfocusedContainerColor = textFieldColor,
+      cursorColor = primaryColor,
+    )
+
+  return Triple(saveButtonColors, cancelButtonColor, textFieldColors)
+}
+
+@Composable
+fun LoiNameDialog(
+  fragment: AbstractFragment,
+  textFieldValue: String,
+  onConfirmRequest: () -> Unit,
+  onDismissRequest: () -> Unit,
+  onTextFieldChange: (String) -> Unit
+) {
+  val (saveButtonColors, cancelButtonColor, textFieldColors) = fragment.getElementColors()
+  AlertDialog(
+    onDismissRequest = onDismissRequest,
+    icon = {},
+    title = {
+      Column(modifier = Modifier.fillMaxWidth()) {
+        Text(
+          text = fragment.getString(R.string.loi_name_dialog_title),
+          fontSize = 5.em,
+          textAlign = TextAlign.Start,
+        )
+      }
+    },
+    text = {
+      Column {
+        Text(
+          text = fragment.getString(R.string.loi_name_dialog_body),
+          modifier = Modifier.padding(0.dp, 0.dp, 0.dp, 16.dp),
```

Comment:
rather than going for a modifier, we can use a separate component called `Spacer` to give a height of 16.dp to bring space between these two components.

Reply:
Good call, done.